### PR TITLE
Use our api to set info on python scrapers

### PR DESCRIPTION
**Pull request by phate89**
This is a splitup from https://github.com/xbmc/xbmc/pull/12661
I replaced the use of properties with our functions we already use to set info in plugins.
I also added nfo url support that @notspiff original implementation was missing.
I'm not sure if it's worth it in music part since there are only few functions available.
@ronie @DaveTBlake any opinion?

## Motivation and Context
We can use functions we are more familiar with.

## How Has This Been Tested?
Tested with my themoviedb python scraper

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by DaveTBlake** [comment]
>I'm not sure if it's worth it in music part since there are only few functions available.

TBH I'm not sure it helps either. For music scraping from NFO, or online, is about getting additional data, hence most of it does not have an equivalent field in CMusicInfoTag. When there is more data to scrape (e.g. I hope to add gender, type and disambiguation to artist) it won't get added to CMusicInfoTag. 

The demo album and artist scrapers are also not how music scraping will work (I have worked on the scrapers you started @ronie but got distracted).

So thank you for considering music @phate89 but I would probably drop that part of it.

File: addons/metadata.demo.tv/demo.py [hunk 1]
```diff
@@ -46,156 +46,146 @@ def get_params():
...
+                     'tag': ['Porn', 'Umomz']
+                    })
+        liz.setRating("imdb", 9, 100000, True )
+        liz.setRating("tvdb", 8.9, 1000)
+        liz.setUniqueIDs({ 'tvdb': '3894', 'imdb' : 'tt384940' }, 'tvdb')
+        liz.addSeason(1, 'Horrible')
+        liz.addSeason(2, 'Crap')
+        liz.setCast([{'name': 'spiff', 'role': 'himself', 'thumbnail': '/home/akva/Pictures/fish.jpg', 'order': 2},
+                    {'name': 'monkey', 'role': 'orange', 'thumbnail': '/home/akva/Pictures/coffee.jpg', 'order': 1}])
+        liz.addAvailableThumb('DefaultBackFanart.png', '1.78')
```
Comment (rmrector) [inline addons/metadata.demo.tv/demo.py]: Like the TV show itself above, the second parameter to `addAvailableThumb` should be an art type, generally 'thumb', or could even something wild like 'clearlogo'.

**Review by rmrector — CHANGES_REQUESTED** [review]
Ohh, these Python scrapers are looking wonderful.

There are a couple of small changes needed, and a suggestion to smooth out a part of the API for artwork.

**Comment by MartijnKaijser** [comment]
@phate89 @notspiff can we throw this one in and start testing?

File: addons/metadata.demo.movies/demo.py [hunk 2]
```diff
@@ -46,71 +46,53 @@ def get_params():
...
+        liz.setRating("themoviedb", 8.9, 1000)
+        liz.setUniqueIDs({ 'imdb': 'tt8938399', 'tmdb' : '9837493' }, 'imdb')
+        liz.setCast([{'name': 'spiff', 'role': 'himself', 'thumbnail': '/home/akva/Pictures/fish.jpg', 'order': 2},
+                    {'name': 'monkey', 'role': 'orange', 'thumbnail': '/home/akva/Pictures/coffee.jpg', 'order': 1}])
+        liz.addAvailableThumb('DefaultBackFanart.png', 'banner')
+        liz.addAvailableThumb('/home/akva/Pictures/hawaii-shirt.png', 'poster')
+        liz.setAvailableFanart([{'image': 'DefaultBackFanart.png', 'preview': 'DefaultBackFanart.png'}, 
+                                {'image': '/home/akva/Pictures/hawaii-shirt.png', 'preview': '/home/akva/Pictures/hawaii-shirt.png'}])
         xbmcplugin.setResolvedUrl(handle=int(sys.argv[1]), succeeded=True, listitem=liz)
 elif action == 'getartwork':
```
[the remark this answers is not in the document]
Comment (phate89) [reply addons/metadata.demo.movies/demo.py]: The names are the same we use in the xml scraper.. That's probably the reason spiff used those..

**Comment by phate89** [comment]
thanks @rmrector for the review..I fixed a couple of your points..
About naming I agree with your naming considerations but the problem is that it's all implemented like it was for xml scraper. addAvailableThumb and setAvailableFanart are really different and they are implemented very differently in kodi core so I preferred not to touch anything and replicate them on python

File: addons/metadata.demo.tv/demo.py (hunk 1 above)
Comment (rmrector) [reply addons/metadata.demo.tv/demo.py]: The aspect in the line above and below still need adjusted, but otherwise this PR looks good.

**Comment by phate89** [comment]
I rebased and renamed some argument like @rmrector suggested. It should be good to go

**Comment by rmrector** [comment]
Looks good, and I definitely appreciate that you were able to rename that guy. Thanks!

**Comment by anthonyryan1** [comment]
I just thought I'd throw out a vote of support for this. I just finished writing a python scraper (not yet published) and this is a significant improvement.

**Pull request merged**